src/components/include/cos_defkernel_api.h

```diff
@@ -40,10 +40,14 @@ struct cos_aep_info {
 
 /* Default Component information */
 struct cos_defcompinfo {
+	spdid_t id;
 	struct cos_compinfo ci;
 	struct cos_aep_info sched_aep;
 };
 
+/* ugly hack to get resmgr to work with current sl interface */
```

Comment:
I have not changed to use unions in `struct cos_defcompinfo` yet for `spdid_t id` ,  `struct cos_compinfo ci`.